src/Microsoft.Azure.WebJobs.Host/Executors/JobHostContextFactory.cs

```diff
@@ -82,6 +82,8 @@ public JobHostContextFactory(IStorageAccountProvider storageAccountProvider, ICo
                 hostIdProvider = new DynamicHostIdProvider(storageAccountProvider, () => functionIndexProvider);
             }
 
+            AzureStorageDeploymentValidator.Validate();
```

Comment:
I've verified that this check fails if Microsoft.Data.Services.Client.dll is not present.